earth_enterprise/rpms/opengee-fusion/post-install.sh

```diff
@@ -0,0 +1,216 @@
+
+    check_fusion_master_or_slave
+
+    install_or_upgrade_asset_root
+
+    final_assetroot_configuration
+
+    final_fusion_service_configuration
+}
+
+#-----------------------------------------------------------------
+# Post-install Functions
+#-----------------------------------------------------------------
+
+setup_fusion_daemon()
+{
+    # setup fusion daemon
+    echo "Setting up the Fusion daemon..."
+
+    [ -f "$INITSCRIPT_UPDATE" ] && "$INITSCRIPT_UPDATE" -f gefusion remove
+    [ -f "$INITSCRIPT_UPDATE" ] && "$INITSCRIPT_UPDATE" gefusion start 90 2 3 4 5 . stop 10 0 1 6 .
+    [ -f "$CHKCONFIG" ] && "$CHKCONFIG" --add gefusion
+
+    echo "Fusion daemon setup ... Done"
+}
+
+create_system_main_directories()
+{
+    mkdir -p "$ASSET_ROOT"
+    mkdir -p "$SOURCE_VOLUME"
+}
+
+compare_asset_root_publishvolume()
+{    
+    if [ -f "$BASEINSTALLDIR_OPT/gehttpd/conf.d/stream_space" ]; then
+        PUBLISH_ROOT_VOLUME="$(cut -d' ' -f3 /opt/google/gehttpd/conf.d/stream_space | cut -d'/' -f2 | $NEWLINECLEANER)"
+
+        if [ -d "$ASSET_ROOT" ] && [ -d "$PUBLISH_ROOT_VOLUME" ]; then
+            VOL_ASSETROOT=$(df "$ASSET_ROOT" | grep -v ^Filesystem | grep -Eo '^[^ ]+')
+            VOL_PUBLISHED_ROOT_VOLUME=$(df "$PUBLISH_ROOT_VOLUME" | grep -v ^Filesystem | grep -Eo '^[^ ]+')
+        fi
+    fi
+}
+
+check_fusion_master_or_slave()
+{
+    if [ -f "$ASSET_ROOT/.config/volumes.xml" ]; then
+        EXISTING_HOST=$(xml_file_get_xpath "$ASSET_ROOT/.config/volumes.xml" "//VolumeDefList/volumedefs/item[1]/host/text()" | $NEWLINECLEANER)
+
+        case "$EXISTING_HOST" in
+            $HOSTNAME_F|$HOSTNAME_A|$HOSTNAME_S|$HOSTNAME)
+                IS_SLAVE=false
+                ;;
+            *)
+                IS_SLAVE=true
+
+                echo -e "\nThe asset root [$ASSET_ROOT] is owned by another Fusion host:  $EXISTING_HOST"
+                echo -e "Installing $GEEF $GEE_VERSION in Grid Slave mode.\n"
+                ;;
+        esac
+    fi
+}
+
+create_systemrc()
+{
+    cat > "$SYSTEMRC" <<END
+<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
+<Systemrc>
+  <assetroot>$ASSET_ROOT</assetroot>
+  <maxjobs>$NUM_CPUS</maxjobs>
+  <locked>0</locked>
+  <fusionUsername>$GEFUSIONUSER</fusionUsername>
+  <userGroupname>$GEGROUP</userGroupname>
+</Systemrc>
+END
+}
+
+install_or_upgrade_asset_root()
+{
+    if [ -f "$SYSTEMRC" ]; then
+        rm -f "$SYSTEMRC"
+    fi
+
+    # create new systemrc file
+    create_systemrc
+
+    chmod 644 "$SYSTEMRC"
+    chown "$GEFUSIONUSER:$GEGROUP" "$SYSTEMRC"
+
+    if [ ! -d "$ASSET_ROOT/.config" ]; then
+        "$BASEINSTALLDIR_OPT/bin/geconfigureassetroot" --new --noprompt \
+            --assetroot "$ASSET_ROOT" --srcvol "$SOURCE_VOLUME"
+        chown -R "$GEFUSIONUSER:$GEGROUP" "$ASSET_ROOT"
+    else
+        # upgrade asset root -- if this is a master
+        if [ "$IS_SLAVE" == false ]; then
+            # TODO: Verify this logic -- this is what is defined in the
```

Comment:
is the 'TODO' still needed ?

Reply:
Yes, as far as I can tell.

This ToDo doesn't mean much to me.  I prefer a comment that says we call with `--nochown` to do this in that case, and without `--nochown` to do that in this case.

I don't know who wrote the ToDo, or the code.  You're welcome to offer a better comment.